Source/WebInspectorUI/UserInterface/Views/ComputedStyleDetailsPanel.js

```diff
@@ -258,26 +262,47 @@ WI.ComputedStyleDetailsPanel = class ComputedStyleDetailsPanel extends WI.StyleD
 
     // Private
 
-    _renderVariablesStyleSectionGroup(style, groupType, label)
+    _createVariablesStyleSection(variablesGroupType, label)
     {
-        let variablesStyleSection = new WI.ComputedStyleSection(this);
+        let variablesStyleSection = new WI.ComputedStyleSection(this, { variablesGroupType });
         variablesStyleSection.propertyVisibilityMode = WI.ComputedStyleSection.PropertyVisibilityMode.HideNonVariables;
         variablesStyleSection.hideFilterNonMatchingProperties = true;
         variablesStyleSection.addEventListener(WI.ComputedStyleSection.Event.FilterApplied, this._handleEditorFilterApplied, this);
         variablesStyleSection.element.dir = "ltr";
-        variablesStyleSection.style = style;
+        variablesStyleSection.style = this.nodeStyles.computedStyle;
 
         this.addSubview(variablesStyleSection);
 
         let detailsSectionRow = new WI.DetailsSectionRow;
         let detailsSectionGroup = new WI.DetailsSectionGroup([detailsSectionRow]);
-        let detailsSection = new WI.DetailsSection(`computed-style-variables-group-${groupType}`, label, [detailsSectionGroup]);
+        let detailsSection = new WI.DetailsSection(`computed-style-variables-group-${variablesGroupType}`, label, [detailsSectionGroup]);
         detailsSection.addEventListener(WI.DetailsSection.Event.CollapsedStateChanged, this._handleDetailsSectionCollapsedStateChanged, this);
         detailsSectionRow.element.appendChild(variablesStyleSection.element);
         this._variablesRow.element.appendChild(detailsSection.element);
 
         this._detailsSectionByStyleSectionMap.set(variablesStyleSection, detailsSection);
-        this._variablesStyleSectionForGroupTypeMap.set(groupType, variablesStyleSection);
+        this._variablesStyleSectionForGroupTypeMap.set(variablesGroupType, variablesStyleSection);
+    }
+
+    _removeVariablesStyleSection(styleSection)
+    {
+        console.assert(this._detailsSectionByStyleSectionMap.has(styleSection), styleSection, "Expected a ComputedStyleSection for a variable group")
+
+        let detailsSection = this._detailsSectionByStyleSectionMap.get(styleSection);
+
+        // WI.ComputedStyleSection is a view but its element is attached to a WI.DetailsSection which isn't. Reparent it before removing.
+        // FIXME: <https://webkit.org/b/152269> - Web Inspector: Convert DetailsSection classes to use View
+        this.element.appendChild(styleSection.element);
```

Comment:
at this point is the `detailsSection` attached to `this.element`?  if so, then this is unnecessary as `WI.View.prototype.removeSubview` only checks that the `WI.View` is in the subtree, not that it's a direct descendant

Reply:
That's true. The DOM juggling isn't necessary anymore.